test/MarketMaker.js

```diff
@@ -108,78 +98,68 @@ contract('marketMaker', function(accounts) {
+    await increaseTimeTestRPC(duration.days(30));
     assert.equal(4, parseInt(await mm.getCurrentPeriodIndex()) );
   });
 
   it("should return correct periods after pausing/unpausing using getCurrentPeriodIndex", async function() {
-    token = await help.simulateCrowdsale(100, [40,30,20,10,0], accounts);
-    const startBlock = web3.eth.blockNumber+10;
-    const blocksPerPeriod = 12;
-
-    mm = await LifMarketMaker.new(token.address, startBlock, blocksPerPeriod, 24,
-      accounts[1], {from: accounts[0]});
+    mmInitialBalance = 20000000;
+    totalTokenSupply = 100;
+    rate = totalTokenSupply / web3.fromWei(mmInitialBalance+10000000, 'ether');
+    crowdale = await help.simulateCrowdsale(rate, [totalTokenSupply], accounts, 1);
+    token = LifToken.at( await crowdale.token.call());
+    mm = LifMarketMaker.at( await crowdale.marketMaker.call());
+    await mm.calculateDistributionPeriods({from: accounts[4]});
 
-    await mm.fund({value: web3.toWei(8, 'ether'), from: accounts[0]});
-    await mm.calculateDistributionPeriods({from: accounts[5]});
+    assert.equal(24, parseInt(await mm.totalPeriods.call()));
 
-    assert.equal(0, parseInt(web3.eth.getBalance(token.address)));
-    assert.equal(startBlock, parseInt(await mm.startBlock()));
-    assert.equal(blocksPerPeriod, parseInt(await mm.blocksPerPeriod()));
-    assert.equal(accounts[1], parseInt(await mm.foundationAddr()));
+    startTimestamp = parseInt(await mm.startTimestamp.call());
 
-    await help.waitToBlock(startBlock);
+    await increaseTimeTestRPCTo(startTimestamp);
     assert.equal(0, parseInt(await mm.getCurrentPeriodIndex()));
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPCTo(startTimestamp+duration.days(30));
     assert.equal(1, parseInt(await mm.getCurrentPeriodIndex()));
     await mm.pause({from: accounts[0]});
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod*3);
+    await increaseTimeTestRPC(duration.days(30)*3);
     await mm.unpause({from: accounts[0]});
     assert.equal(1, parseInt(await mm.getCurrentPeriodIndex()));
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPC(duration.days(30));
     assert.equal(2, parseInt(await mm.getCurrentPeriodIndex()));
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPC(duration.days(30));
     assert.equal(3, parseInt(await mm.getCurrentPeriodIndex()));
     await mm.pause({from: accounts[0]});
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod*2);
+    await increaseTimeTestRPC(duration.days(30)*2);
     await mm.unpause({from: accounts[0]});
-    await help.waitToBlock(web3.eth.blockNumber+blocksPerPeriod);
+    await increaseTimeTestRPC(duration.days(30));
     assert.equal(4, parseInt(await mm.getCurrentPeriodIndex()));
   });
 
   const periods = 24;
-  const tokenTotalSupply = 100;
+  const tokenTotalSupply = 3000;
```

Comment:
why are we changing all the numbers in this test, starting from here?

Reply:
Because the rate was less than one, thing that wont happing and in order to create the MarketMaker form the crowdsale the rate have to be higher than one.